crates/shared/src/price_estimation/trade_verifier/mod.rs

```diff
@@ -234,9 +239,13 @@ impl TradeVerifier {
             }
         };
 
-        let mut summary = SettleOutput::decode(&output?, query.kind, &tokens)
-            .context("could not decode simulation output")
-            .map_err(Error::SimulationFailed)?;
+        let mut summary = SettleOutput::from_swap(
+            output
+                .context("simulation failed")
+                .map_err(Error::SimulationFailed)?,
```

Comment:
This is wrapping the already wrapped error.
```suggestion
            output?,
```